spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonTimeoutAutoConfiguration.java

```diff
@@ -51,63 +53,68 @@ public HandlerTimeoutHandlerEnhancerDefinition messageTimeoutHandlerEnhancerDefi
-
-        private final TimeoutProperties.TransactionTimeoutProperties properties;
+    private record AxonTimeoutConfigurerModule(
+            @Nonnull TimeoutProperties.TransactionTimeoutProperties properties
+    ) implements ConfigurationEnhancer {
 
-        public AxonTimeoutConfigurerModule(TimeoutProperties properties) {
-            this.properties = properties.getTransaction();
+        @Override
+        public int order() {
+            return Integer.MIN_VALUE;
         }
 
         @Override
-        public void configureModule(@Nonnull LegacyConfigurer configurer) {
-            configurer.eventProcessing()
-                      .registerDefaultHandlerInterceptor((c, name) -> {
-                          TaskTimeoutSettings settings = getSettingsForProcessor(name);
-                          return new UnitOfWorkTimeoutInterceptorBuilder(
-                                  "EventProcessor " + name,
-                                  settings.getTimeoutMs(),
-                                  settings.getWarningThresholdMs(),
-                                  settings.getWarningIntervalMs()
-                          ).buildEventInterceptor();
-                      });
-            // Cannot use the configurer.onInitialize, as it creates a circular creation dependency
-            configurer.onStart(Integer.MIN_VALUE, () -> {
-                LegacyConfiguration c = configurer.buildConfiguration();
-                // TODO #3103 - Revisit this section to adjust it to configurer logic instead of configuration logic.
-//                c.commandBus().registerHandlerInterceptor(new UnitOfWorkTimeoutInterceptor(
-//                        c.commandBus().getClass().getSimpleName(),
-//                        properties.getCommandBus().getTimeoutMs(),
-//                        properties.getCommandBus().getWarningThresholdMs(),
-//                        properties.getCommandBus().getWarningIntervalMs()
-//                ));
-//                c.queryBus().registerHandlerInterceptor(new UnitOfWorkTimeoutInterceptorBuilder(
-//                        c.queryBus().getClass().getSimpleName(),
-//                        properties.getQueryBus().getTimeoutMs(),
-//                        properties.getQueryBus().getWarningThresholdMs(),
-//                        properties.getQueryBus().getWarningIntervalMs()
-//                ).buildQueryInterceptor());
-//                c.deadlineManager().registerHandlerInterceptor(new UnitOfWorkTimeoutInterceptorBuilder(
-//                        c.deadlineManager().getClass().getSimpleName(),
-//                        properties.getDeadline().getTimeoutMs(),
-//                        properties.getDeadline().getWarningThresholdMs(),
-//                        properties.getDeadline().getWarningIntervalMs()
-//                ).buildDeadlineInterceptor());
-            });
+        public void enhance(@Nonnull ComponentRegistry registry) {
```

Comment:
It's nice to bring this back, thanks!
I was just wondering, shouldn't we build timeouts into the construction of the unit of work? :thinking: 
as this returns a completablefuture that can be timed out through it's `get`